### PR TITLE
Add support for multiple event handlers

### PR DESCRIPTION
**Pull request by jepcd**

### CODE REVIEW

File: src/client/mod.rs [hunk 1]
```diff
@@ -302,40 +302,43 @@ impl ClientBuilder {
         self.intents
     }
 
-    /// Sets an event handler with multiple methods for each possible event.
-    pub fn event_handler<H: EventHandler + 'static>(mut self, event_handler: H) -> Self {
```
Comment (kangalio) [inline src/client/mod.rs:306]: I think it's better to remove the add_ prefix from `add_event_handler`, `add_event_handler_arc`, and `add_raw_event_handler` again, because it causes needless churn.

File: src/client/mod.rs [hunk 2]
```diff
@@ -302,40 +302,43 @@ impl ClientBuilder {
...
+        self.event_handlers.push(event_handler_arc);
 
         self
     }
 
-    /// Gets the event handler, if already initialized. See [`Self::event_handler`] for more info.
-    pub fn get_event_handler(&self) -> Option<Arc<dyn EventHandler>> {
-        self.event_handler.clone()
+    /// Gets the added event handlers. See [`Self::event_handler`] for more info.
+    pub fn get_event_handlers(&self) -> Vec<Arc<dyn EventHandler>> {
```
Comment (kangalio) [inline src/client/mod.rs]: Probably better to return `&[Arc<dyn EventHandler>]` here now. The caller doesn't necessarily need a cloned Vec, so we should leave the choice to the caller.

Same with get_raw_event_handlers below

File: src/client/mod.rs (hunk 1 above)
Comment (jepcd) [reply src/client/mod.rs:306]: done

File: src/client/mod.rs (hunk 2 above)
Comment (jepcd) [reply src/client/mod.rs]: done

File: README.md [hunk 3]
```diff
@@ -131,51 +131,51 @@ The default features are: `builder`, `cache`, `client`, `framework`, `gateway`,
 
 There are these alternative default features, they require to set `default-features = false`:
 
-- **default_native_tls**: Uses `native_tls_backend` instead of the default `rustls_backend`.
-- **default_no_backend**: Excludes the default backend, pick your own backend instead.
+-   **default_native_tls**: Uses `native_tls_backend` instead of the default `rustls_backend`.
+-   **default_no_backend**: Excludes the default backend, pick your own backend instead.
```
Comment (kangalio) [inline README.md]: Why was this additional whitespace added?

File: README.md [hunk 4]
```diff
@@ -211,17 +211,17 @@ a Rust-native cloud development platform that allows deploying Serenity bots for
...
+-   [Songbird][project:songbird]: An async Rust library for the Discord voice API.
+-   [Poise][project:poise]: Experimental command framework, with advanced features like edit tracking, single function slash and prefix commands and flexible argument parsing.
+
+[`cache`]: https://docs.rs/serenity/*/serenity/cache/struct.Cache.html
+[`client::builder`]: https://docs.rs/serenity/*/serenity/client/struct.Client.html#method.builder
+[`eventhandler::message`]: https://docs.rs/serenity/*/serenity/client/trait.EventHandler.html#method.message
+[`context`]: https://docs.rs/serenity/*/serenity/client/struct.Context.html
+[`event`]: https://docs.rs/serenity/*/serenity/model/event/enum.Event.html
+[`event::messagecreate`]: https://docs.rs/serenity/*/serenity/model/event/enum.Event.html#variant.MessageCreate
+[`shard`]: https://docs.rs/serenity/*/serenity/gateway/struct.Shard.html
```
Comment (kangalio) [inline README.md]: And all of the link keys made lowercase

File: src/client/mod.rs [hunk 5]
```diff
@@ -302,40 +302,40 @@ impl ClientBuilder {
...
     /// events will be dispatched.
     pub fn raw_event_handler<H: RawEventHandler + 'static>(mut self, raw_event_handler: H) -> Self {
-        self.raw_event_handler = Some(Arc::new(raw_event_handler));
+        self.raw_event_handlers.push(Arc::new(raw_event_handler));
 
         self
     }
 
-    /// Gets the raw event handler, if already initialized. See [`Self::raw_event_handler`] for more
+    /// Gets the added raw event handlers. See [`Self::raw_event_handlers`] for more
```
Comment (kangalio) [inline src/client/mod.rs]: Update the intra-doc link to remove the `s`

**Comment by jepcd** [comment]
yikes, I forgot to disable my format on save thing, fixed now

File: README.md [hunk 6]
```diff
@@ -247,4 +247,4 @@ a Rust-native cloud development platform that allows deploying Serenity bots for
 [repo:lavaplayer]: https://github.com/sedmelluq/lavaplayer
 [logo]: https://raw.githubusercontent.com/serenity-rs/serenity/current/logo.png
 [rust-version-badge]: https://img.shields.io/badge/rust-1.64.0+-93450a.svg?style=flat-square
-[rust-version-link]: https://blog.rust-lang.org/2022/09/22/Rust-1.64.0.html
+[rust-version-link]: https://blog.rust-lang.org/2022/09/22/Rust-1.64.0.html
```
Comment (kangalio) [inline README.md]: Still a spurious unrelated change here. Probably just a trailing newline added/removed

**Review by kangalio — APPROVED** [review]

File: README.md (hunk 6 above)
Comment (jepcd) [reply README.md]: should be gone now

**Review by kangalio — APPROVED** [review]

**Review by arqunis — APPROVED** [review]
